### PR TITLE
feat: add patch endpoint to update enterprise customer catalog

### PR DESCRIPTION
**Pull request by katrinan029**
# Description
We need an endpoint that supports updating the title of an enterprise customer catalog.

# Solution
Added a PATCH endpoint to url `enterprise_customer_catalog` to support the PATCH request.

[JIRA](https://2u-internal.atlassian.net/browse/ENT-7583)

This PR is part of an epic to implement an edit/view learner credit plan details in support tools. Below are the previous tickets associated:

|Implementation|PR|
| ------------- | ------------- |
|Add feature flag| https://2u-internal.atlassian.net/browse/ENT-7517| 
|Create List View of Learner Credit Plans with Muted Standalone Hyperlinks|https://github.com/openedx/frontend-app-support-tools/pull/349|
|Implements Detailed Plan View with Edit Mode Button|https://github.com/openedx/frontend-app-support-tools/pull/352|

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/openedx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/openedx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/openedx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/openedx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/api/v1/views/enterprise_customer_catalog.py [hunk 1]
```diff
@@ -76,6 +82,32 @@ def create(self, request, *args, **kwargs):
...
+            "uuid": string - UUID of an existing enterprise customer catalog
+            "title":  string - Title of the catalog,
+          }
+
+        Returns 200 along with the updated object.
+        """
+        enterprise_customer_catalog_uuid = request.data.get('uuid')
+        found_catalog = self.has_enterprise_customer_catalog(enterprise_customer_catalog_uuid)
+        if not found_catalog:
+            return Response({'detail': 'Could not find catalog uuid'}, status=status.HTTP_404_NOT_FOUND)
```
Comment (kiram15) [inline enterprise/api/v1/views/enterprise_customer_catalog.py:104]: I would put the uuid that wasn't found in the response

File: tests/test_enterprise/api/test_views.py [hunk 2]
```diff
@@ -1876,10 +1876,59 @@ def test_create_catalog_incorrect_data(self):
...
+        patch_response_output = self.load_json(patch_response.content)
+
+        assert patch_response.status_code == 404
+        assert "Could not find catalog uuid" in patch_response_output['detail']
+
 
 @ddt.ddt
 @mark.django_db
-class TestEntepriseCustomerCatalogs(BaseTestEnterpriseAPIViews):
+class TestEnterpriseCustomerCatalogs(BaseTestEnterpriseAPIViews):
```
Comment (kiram15) [inline tests/test_enterprise/api/test_views.py:1931]: Good catch!

**Review by kiram15 — APPROVED** [review]

File: enterprise/api/v1/views/enterprise_customer_catalog.py (hunk 1 above)
Comment (katrinan029) [reply enterprise/api/v1/views/enterprise_customer_catalog.py]: good call!

File: tests/test_enterprise/api/test_views.py [hunk 3]
```diff
@@ -1876,10 +1876,59 @@ def test_create_catalog_incorrect_data(self):
...
+
+        self.set_jwt_cookie(ENTERPRISE_ADMIN_ROLE, str(enterprise_customer.uuid))
+        self.user.is_staff = True
+        self.user.save()
+
+        post_response = self.client.post(ENTERPRISE_CUSTOMER_CATALOG_ENDPOINT, {
+            "title": "Test Catalog",
+            "enterprise_customer": str(enterprise_customer.uuid),
+            "enterprise_catalog_query": str(enterprise_catalog_query.id),
+        }, format='json')
```
[the remark this answers is not in the document]
Comment (katrinan029) [reply tests/test_enterprise/api/test_views.py:1894]: I agree

**Pull request merged**